x-pack/plugins/security_solution/server/lib/detection_engine/signals/utils.ts

```diff
@@ -57,10 +63,81 @@ export const shorthandMap = {
   },
 };
 
+export const hasReadIndexPrivileges = async (
+  privileges: Privilege,
+  logger: Logger,
+  buildRuleMessage: BuildRuleMessage,
+  ruleStatusService: RuleStatusService
+): Promise<boolean> => {
+  const indexNames = Object.keys(privileges.index);
+  const [indexesWithReadPrivileges, indexesWithNoReadPrivileges] = partition(
+    indexNames,
+    (indexName) => privileges.index[indexName].read
+  );
+
+  if (indexesWithReadPrivileges.length > 0 && indexesWithNoReadPrivileges.length > 0) {
+    // some indices have read privileges others do not.
+    // set a partial failure status
+    const errorString = `Missing required read permissions on indexes: ${JSON.stringify(
+      indexesWithNoReadPrivileges
+    )}`;
+    logger.error(buildRuleMessage(errorString));
+    await ruleStatusService.partialFailure(errorString);
+    return true;
+  } else if (
+    indexesWithReadPrivileges.length === 0 &&
+    indexesWithNoReadPrivileges.length === indexNames.length
+  ) {
+    // none of the indices had read privileges so set the status to failed
+    // since we can't search on any indices we do not have read privileges on
+    const errorString = `The rule does not have read privileges to any of the following indices: ${JSON.stringify(
+      indexesWithNoReadPrivileges
+    )}`;
+    logger.error(buildRuleMessage(errorString));
+    await ruleStatusService.error(errorString);
+    return true;
+  }
+  return false;
+};
+
+export const hasTimestampFields = async (
+  wroteStatus: boolean,
+  timestampField: string,
+  // any is derived from here
+  // node_modules/@elastic/elasticsearch/api/kibana.d.ts
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  timestampFieldCapsResponse: ApiResponse<Record<string, any>, Context>,
+  ruleStatusService: RuleStatusService,
+  logger: Logger,
+  buildRuleMessage: BuildRuleMessage
+): Promise<boolean> => {
+  if (
+    !wroteStatus &&
+    timestampField != null &&
```

Comment:
`timestampField` is typed as `string` above. Is the type incorrect or is this check unnecessary?

Reply:
ah yeah that predicate was leftover from when I moved this into a util. Don't need that anymore! thanks 👍